src/Symfony/Contracts/HttpClient/ChunkInterface.php

```diff
@@ -47,6 +47,15 @@ public function isFirst(): bool;
      */
     public function isLast(): bool;
 
+    /**
+     * Tells when an 1xx status code was just received.
+     *
+     * @param array|null &$headers Set by reference to the headers of the informational response
+     *
+     * @throws TransportExceptionInterface on a network error or when the idle timeout is reached
+     */
+    public function isInformational(?array &$headers = []): int;
```

Comment:
Why can it be nullable? And Why not returning the headers instead of using a reference?

Reply:
> Why can it be nullable?

So that one can call the method with an uninitialized variable, as done in e.g. tests. Not doing so would force initializing the variable to the empty array before calling isInformational.

> And Why not returning the headers instead of using a reference?

because the return value is the status code - and not all status code have headers.

Reply:
yes but it could be something like ['code' => 103, 'headers' => []) (headers optional), or even better: an instance of `ResponseInterface` (with no headers if there are no headers).

Reply:
> it could be something like ['code' => 103, 'headers' => []) (headers optional)

you'd prefer this? that'd work also that's true:
`if ([$status, $headers] = $chunk->isInformational()) {...`

> or even better: an instance of ResponseInterface (with no headers if there are no headers).

that would be very confusing as there is already a response, and all other methods would make no sense. This doesn't fit semantically also, this is not a response, but an informational status code.

Reply:
It's actually an HTTP response, according to the RFC:

> This memo defines a status code for sending an informational **response**
   ([RFC7231], Section 6.2) that contains header fields that are likely
   to be included in the final response.

Reply:
Also in the original RFC 2616:

> 10.1 Informational 1xx
>   This class of status code indicates a provisional response,
   consisting only of the Status-Line and optional headers, and is
   terminated by an empty line. There are no required headers for this
   class of status code. Since HTTP/1.0 did not define any 1xx status
   codes, servers MUST NOT send a 1xx response to an HTTP/1.0 client
   except under experimental conditions.

It would make sens to return a response object, with an empty body. (Similar to what you expect when using `HEAD` for instance).

Reply:
And it has been clarified in HTTPBis (RFC 7231) (emphasize mine):

>    The 1xx (Informational) class of status code indicates an interim
   response for communicating connection status or request progress
   prior to completing the requested action and sending a final
   response. 1xx responses are terminated by the first empty line after
   the status-line (the empty line signaling the end of the header
   section).  Since HTTP/1.0 did not define any 1xx status codes, a
   server MUST NOT send a 1xx response to an HTTP/1.0 client.

>   **A client MUST be able to parse one or more 1xx responses received
   prior to a final response, even if the client does not expect one.**  A
   user agent MAY ignore unexpected 1xx responses.

Reply:
All this in the RFC is all implemented with the PR as is. I'm definitely sure an instance of ResponseInterface would not fit here. There are too many semantic mismatches. We could introduce another interface to return a corresponding implementation here, but honestly this is not worth it to me.
int+array cover all possible data that the protocol could define.

Also, I prefer the current method that returns an int, as it allows checking the value of the code is one line.

Thanks for challenging :)

Reply:
the signature is now `getInformationalStatus(): ?array`, returning a `[$statusCode, $headers]` tuple